src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -447,7 +448,7 @@ private static MoveValidationResult validateNonCombat(final GameData data, final
     }
     // if there are enemy units on the path blocking us, that is validated elsewhere (validateNonEnemyUnitsOnPath)
     // now check if we can move over neutral or enemies territories in noncombat
-    if (Match.allMatchNotEmpty(units, Matches.UnitIsAir)
+    if (!units.isEmpty() && Match.allMatch(units, Matches.UnitIsAir)
```

Comment:
Parentheses are not required here, but it may make it easier to read (I got hung up on the `||` for a few minutes).